R/intervals.R

```diff
@@ -0,0 +1,232 @@
+
+    # Make start from last dose (pknca_dose) or first concentration (pknca_conc)
+    mutate(start = if (start_from_last_dose) !!sym(time_column)
+           else !!sym(time_column) + !!sym("ARRLT")) %>%
+    group_by(!!!syms(conc_groups)) %>%
+    arrange(start) %>%
+
+    # Make end based on next dose time (if no more, TRTRINT or last NFRLT)
+    mutate(end = if (has_trtrint) {
+      case_when(
+        !is.na(lead(!!sym(time_column))) ~ lead(!!sym(time_column)),
+        is.na(TRTRINT) & is_one_dose ~ Inf,
+        is.na(TRTRINT) ~ start + max_end,
+        TRUE ~ start + TRTRINT
+      )
+    } else {
+      case_when(
+        !is.na(lead(!!sym(time_column))) ~ lead(!!sym(time_column)),
+        is_one_dose ~ Inf,
+        TRUE ~ start + max_end
+      )
+    }
+    ) %>%
+    ungroup() %>%
+    select(any_of(c("start", "end", conc_groups,
+                    "ATPTREF", "DOSNOA", "VOLUME"))) %>%
+
+    # Create logical columns with only TRUE for the NCA parameters requested by the user
+    mutate(!!!setNames(rep(FALSE, length(params)), params)) %>%
+    # Identify the intervals as the base ones for the NCA analysis
+    mutate(type_interval = "main")
+
+}
+
+#' Update an intervals data frame with user-selected parameters by study type
+#'
+#' @param data A PKNCAdata object containing intervals and dosing data.
+#' @param parameter_selections A named list of selected PKNCA parameters by study type.
+#' @param study_types_df A data frame mapping analysis profiles to their study type.
+#' @param auc_data A data frame containing partial AUC ranges.
+#' @param impute Logical indicating whether to impute start values for parameters.
+#'
+#' @returns An updated PKNCAdata object with parameter intervals based on user selections.
+#'
+update_main_intervals <- function(data, parameter_selections,
+                                  study_types_df, auc_data, impute = TRUE) {
+
+  all_pknca_params <- setdiff(names(PKNCA::get.interval.cols()), c("start", "end"))
+
+  # Determine the grouping columns from the study_types_df
+  grouping_cols <- setdiff(names(study_types_df), c("type"))
+  missing_columns <- setdiff(grouping_cols, colnames(data$intervals))
+  # check for grouping cols in intervals
+  if (length(missing_columns) > 0) {
+    stop(paste("Missing required columns:", paste(missing_columns, collapse = ", ")))
+  }
+
+  # 1. Add the 'type' column to the intervals data
+  intervals_with_types <- data$intervals %>%
+    left_join(study_types_df, by = grouping_cols)
+
+  # 3. Iterate over selections to set new flags to TRUE
+  updated_intervals <- intervals_with_types
+  if (length(parameter_selections) > 0) {
+    for (study_type in names(parameter_selections)) {
+      params_for_type <- parameter_selections[[study_type]]
+      if (length(params_for_type) > 0) {
+        updated_intervals <- updated_intervals %>%
+          mutate(across(
+            .cols = all_of(params_for_type),
+            .fns = ~ if_else(type == study_type, TRUE, .)
```

Comment:
issue(non-blocking): Please use anonymous functions instead of formulas if possible.
```suggestion
            .fns = \(x) if_else(type == study_type, TRUE, x)
```